prime-router/docs/design/proposals/0024-original-message-passthrough.md

```diff
@@ -0,0 +1,172 @@
+      modified.
+    * *Con*: This approach is highly reliant on the processes performed at the `receive` pipeline step remaining as they
+      are now. Any refactoring of this code must take sendOriginal into consideration.
+
+## Limitations
+
+* Filtering can still occur as a routing process, but the nature of sending an unmodified HL7v2 file will mean any
+  conditions that would have been eliminated by filtering would be sent regardless.
+* Batching cannot be performed for sendOriginal files.
+* Filenames in the Azure blobstore currently reflect pipeline specific information to more easily identify when and why
+  a file was created.
+    * There is currently a limitation that files must have unique names, and blobURLs cannot be reused.
+    * External filenames are in many cases based on the blobURL presently.
+
+## Feature Design
+
+After performing analysis of the possible implementations above, the **Add feature to the UP translate step: retrieve
+blobURL from report_file DB table** implementation was selected.
+
+The overall design of the sendOriginal feature is as follows:
+
+* A new `Topic` feature flag, `sendOriginal`, will be used for sendOriginal items. The use of Topic to define
+  sendOriginal guarantees that data that must be handled via sendOriginal is not processed as other
+  full ELR data and is only sent to receivers in its original state. This also allows receivers to be able to configure
+  endpoints specific to the data for this particular sender, separate from any batching settings the receiver may
+  otherwise have configured.
+* As the `Topic` is unique, no other identifiers for sendOriginal items are needed. Configuring the sender and
+  receivers for the Topic with `sendOriginal` defined is all that is required.
+* The `translate` step will copy the original blob data from the blobstore instead of translating from the internal
+  FHIR bundle. Enhancements to `BlobAccess` that have been made in preparation for the `Manage Translation Schemas`
+  feature can be utilized to perform this operation. The `translate` step will then directly insert the resulting blob
+  to the `send` queue.
+* `Report.formExternalFilename` uses the filename of the blobstore item, so there should not be any further
+  modifications necessary as long as the blobstore item is named correctly by the `translate` function.
+
+### Issues to be created
+
+* Add header to `receive` step to allow sender to supply original filename and store in database
+  field `report_file.external_name`
+* Write function to look up original filename and blob URL for a given report ID
+* Add step to `translate` function to check message topic, copy the original blob with original filename (with report ID
+  appended), and insert into send queue
+    * The database currently requires blobURLs to be unique. Appending the report ID to the original filename should
```

Comment:
I want to be clear that if a receiver gets a file from RS that has the ReportID included at the start or end of the original file name that is acceptable.